update: Revamp full disk encryption section (#2437)
privacyguides/privacyguides.org: docs/encryption.md

## OS Full Disk Encryption

> Full disk encryption (FDE) is a comprehensive data encryption solution, encompassing the operating system and system files. FDE principally leverages hardware security features, such as a [secure cryptoprocessor](https://en.wikipedia.org/wiki/Secure_cryptoprocessor). Therefore, we recommend using the built-in FDE solutions for your operating system. For external drives, however, we still recommend open-source tools for additional flexibility and to avoid vendor lock-in.

Reviewer: Perhaps mentioning it in a shorter fashion would be better since the sentence is already on the edge of being verbose. Maybe "(e.g., a TPM)"

> Full disk encryption (FDE) is a comprehensive data encryption solution, encompassing the operating system and system files. FDE principally leverages hardware security features, such as a [secure cryptoprocessor](https://en.wikipedia.org/wiki/Secure_cryptoprocessor). Therefore, we recommend using the built-in FDE solutions for your operating system. For external drives, however, we still recommend open-source tools for additional flexibility and to avoid vendor lock-in.

Reviewer 2: Here are the corrected changes with your suggestion: 

```suggestion
Full disk encryption (FDE) is a comprehensive data encryption solution, encompassing the operating system and system files. FDE principally leverages hardware security features, such as a [secure cryptoprocessor](https://en.wikipedia.org/wiki/Secure_cryptoprocessor) (e.g., a [TPM](https://en.wikipedia.org/wiki/Trusted_Platform_Module)). Therefore, we recommend using the built-in FDE solutions for your operating system. For external drives, however, we still recommend open-source tools for additional flexibility and to avoid vendor lock-in.
```

- You can change your preboot pin at any time by running: `manage-bde -changepin c:`, again, replacing `c:` with the drive letter of your operating system drive.
- To remove the preboot pin:
    1. Configure the Group Policy "**Configure TPM Startup PIN**" back to "**Allow Startup PIN With TPM**".
    2. Run: `manage-bde -protectors -add c: -TPM`.

</details>

While BitLocker is not officially supported on Windows Home, it can be enabled on Home editions with a few extra steps.

<details class="example" markdown>
<summary>Enabling BitLocker on Windows Home</summary>

#### Preliminary checks

- You must have partitions formatted with a [GUID Partition Table](https://en.wikipedia.org/wiki/GUID_Partition_Table).
- You must have a TPM (v1.2, 2.0+) and it must be enabled.
- You may need to [disable the non-Bitlocker "Device encryption" functionality](https://discuss.privacyguides.net/t/enabling-bitlocker-on-the-windows-11-home-edition/13303/5) if it is enabled.
- This guide assumes your computer has a typical disk configuration. If you have a non-standard configuration, such as a dual-boot setup, you may want to avoid this workaround.

> - This guide also assumes your operating system drive letter is "**C**". If it is not, you may need to replace `c:` with the correct drive letter.

Reviewer 3: ```suggestion
+ If your computer has multiple drives, you will need to take a few extra steps to determine the correct drive letter, because it is **not** necessarily the same between when your operating system is booted and while in the recovery environment (see step 3 below).
+ <details class="info" markdown>
+ <summary>Determining the correct drive letter</summary>
+ 1. Run, and accept the elevation prompt:
+      ```powershell
+      diskpart
+      ```
+ 2. In the new window that appears, run:
+      ```powershell
+      list volume
+      ```
+      Remember the volume number your operating system boots on (check the "ltr" column).
+ 3. Follow the below instructions until and including step 3.
+ 4. Run the commands in the above steps 1 and 2 again.
+ 5. For steps 4-7 in the below instructions, use the drive letter that corresponds with the aforementioned volume number.
+ </details>
```

Reviewer: Thanks for writing out these instructions

## OS Full Disk Encryption

> Full disk encryption (FDE) is a comprehensive data encryption solution, encompassing the operating system and system files. FDE principally leverages hardware security features, such as a [secure cryptoprocessor](https://en.wikipedia.org/wiki/Secure_cryptoprocessor) (e.g., a [TPM](https://en.wikipedia.org/wiki/Trusted_Platform_Module)). Therefore, we recommend using the built-in FDE solutions for your operating system. For external drives, however, we still recommend open-source tools for additional flexibility and to avoid vendor lock-in.

Reviewer 4: For the last sentence in the introductory paragraph, I'm not sure source availability is _that_ relevant here. I think that cross-/multi-platform might be more apt with respect to the considerations mentioned ("for additional flexibility and to avoid vendor lock-in").
```suggestion
Full disk encryption (FDE) is a comprehensive data encryption solution, encompassing the operating system and system files. FDE principally leverages hardware security features, such as a [secure cryptoprocessor](https://en.wikipedia.org/wiki/Secure_cryptoprocessor) (e.g., a [TPM](https://en.wikipedia.org/wiki/Trusted_Platform_Module)). Therefore, we recommend using the built-in FDE solutions for your operating system. For external drives, however, we still recommend [cross-platform tools](#multi-platform) for additional flexibility and to avoid vendor lock-in.
```
There's already a section for cross-platform tools on this recommendation page anyway, so I embedded an internal link.

> Full disk encryption (FDE) is a comprehensive data encryption solution, encompassing the operating system and system files. FDE principally leverages hardware security features, such as a [secure cryptoprocessor](https://en.wikipedia.org/wiki/Secure_cryptoprocessor) (e.g., a [TPM](https://en.wikipedia.org/wiki/Trusted_Platform_Module)). Therefore, we recommend using the built-in FDE solutions for your operating system. For external drives, however, we still recommend open-source tools for additional flexibility and to avoid vendor lock-in.

Reviewer 4: For the last sentence in the introductory paragraph, I'm not sure source availability is _that_ relevant here. I think that cross-/multi-platform might be more apt with respect to the considerations mentioned ("for additional flexibility and to avoid vendor lock-in").
```suggestion
Full disk encryption (FDE) is a comprehensive data encryption solution, encompassing the operating system and system files. FDE principally leverages hardware security features, such as a [secure cryptoprocessor](https://en.wikipedia.org/wiki/Secure_cryptoprocessor) (e.g., a [TPM](https://en.wikipedia.org/wiki/Trusted_Platform_Module)). Therefore, we recommend using the built-in FDE solutions for your operating system. For external drives, however, we still recommend [cross-platform tools](#multi-platform) for additional flexibility and to avoid vendor lock-in.
```
There's already a section for cross-platform tools on this recommendation page anyway, so I embedded an internal link.

## OS Full Disk Encryption

Full disk encryption (FDE) is a comprehensive data encryption solution, encompassing the operating system and system files. FDE principally leverages hardware security features, such as a [secure cryptoprocessor](https://en.wikipedia.org/wiki/Secure_cryptoprocessor). Therefore, we recommend using the built-in FDE solutions for your operating system. For external drives, however, we still recommend open-source tools for additional flexibility and to avoid vendor lock-in.

Note the terms full *disk* encryption and full *volume* encryption are often used interchangeably.

### BitLocker

<div class="admonition recommendation" markdown>

![BitLocker logo](assets/img/encryption-software/bitlocker.png){ align=right }

> Built into Windows and a proprietary solution, **BitLocker** provides full volume encryption to protect data at rest. BitLocker [leverages](https://learn.microsoft.com/en-us/windows/security/hardware-security/tpm/how-windows-uses-the-tpm#tpm-overview) the Trusted Platform Module (TPM) for hardware-based security, provided one is present. BitLocker is only officially supported on Pro, Enterprise and Education editions of Windows.

Reviewer 2: ```suggestion
**BitLocker**, a proprietary solution built into Windows, provides full volume encryption to protect data at rest. BitLocker [leverages](https://learn.microsoft.com/en-us/windows/security/hardware-security/tpm/how-windows-uses-the-tpm#tpm-overview) the Trusted Platform Module (TPM) for hardware-based security, provided one is present in your device. BitLocker is only officially supported on Pro, Enterprise and Education editions of Windows.

Reviewer 2: This will be a better wording.

Reviewer 5: Thanks, that is more consistent with the rest of the site.

Reviewer 2: Hmm. But that's doesn't seem what I asked about. Please follow my 'suggested change'

    **Computer Configuration** → **Administrative Templates** → **Windows Components** → **BitLocker Drive Encryption** → **Operating System Drives**
    2. Open and enable the setting "**Require Additional Authentication at Startup**."
    3. Under "**Configure TPM Startup PIN**," select: "**Require Startup PIN With TPM**."
3. Add a pin:
    1. As an administrator, open a command prompt window.
    2. Type and run:<br>
        `manage-bde -protectors -add c: -TPMAndPIN`<br>
        If necessary, replace `c:` with the drive letter of your operating system drive.
    3. Type and confirm a pin of 6 to 20 characters (you won't see the characters as you type them).
4. Check the preboot authentication status:
    1. Type and run: `manage-bde -status`.
    2. Check whether "**TPM and Pin**" is listed under "**Key Protectors**."

- You can change your preboot pin at any time by running: `manage-bde -changepin c:`, again, replacing `c:` with the drive letter of your operating system drive.
- To remove the preboot pin:
    1. Configure the Group Policy "**Configure TPM Startup PIN**" back to "**Allow Startup PIN With TPM**".
    2. Run: `manage-bde -protectors -add c: -TPM`.

</details>

> While BitLocker is not officially supported on Windows Home, it can be enabled on Home editions with a few extra steps.

Reviewer 5: Good question, I did see that issue. To me it read as if it worked except they accidentally did the process on a non-boot drive since their drive letters were weird.

Which is why I added:
```diff
+ This guide assumes the drive letter of your operating system drive is "C". If it is not, replace `c:` with the correct drive letter in the following commands.
```

Though maybe I'm interpreting what they're saying wrong.

> While BitLocker is not officially supported on Windows Home, it can be enabled on Home editions with a few extra steps.

Reviewer 6: There was also a thread https://discuss.privacyguides.net/t/enabling-bitlocker-on-the-windows-11-home-edition/13303/ about it. could just check in a VM

> While BitLocker is not officially supported on Windows Home, it can be enabled on Home editions with a few extra steps.

Reviewer 5: I went through the guide on a VM and it worked for me. I updated the instructions based on what I experienced (i.e., saving the recovery key to a .txt file didn't work for me).

![Device encryption enabled in a Windows 11 Home virtual machine.](https://github.com/privacyguides/privacyguides.org/assets/140852203/b9e3c454-cb7e-4d36-b5e0-26d423168fac)

> While BitLocker is not officially supported on Windows Home, it can be enabled on Home editions with a few extra steps.

Reviewer 3: or we could just point home users to MAS.

> While BitLocker is not officially supported on Windows Home, it can be enabled on Home editions with a few extra steps.

Reviewer: > The guide worked, my problem was that the drive letters were _switched_ in the troubleshooter command prompt. I think the addition should explicitly mention the possibility of this.

Where should this be mentioned? It's been a few months since I went through the guide, so I'm admittedly not too familiar with it anymore.

> or we could just point home users to MAS.

MAS works well — there's no denying that. Though I think it's more of a iykyk thing rather than something to be recommend.

> While BitLocker is not officially supported on Windows Home, it can be enabled on Home editions with a few extra steps.

Reviewer 3: > MAS works well — there's no denying that. Though I think it's more of a iykyk thing rather than something to be recommend.

Is what we're suggesting any less illegal?

> ## OS Full Disk Encryption

Reviewer 7: ```suggestion
## OS Encryption
```

> Full disk encryption (FDE) is a comprehensive data encryption solution, encompassing the operating system and system files. FDE principally leverages hardware security features, such as a [secure cryptoprocessor](https://en.wikipedia.org/wiki/Secure_cryptoprocessor) (e.g., a [TPM](https://en.wikipedia.org/wiki/Trusted_Platform_Module)). Therefore, we recommend using the built-in FDE solutions for your operating system. For external drives, however, we still recommend [cross-platform tools](#multi-platform) for additional flexibility and to avoid vendor lock-in.

Reviewer 8: Nothing about FDE inherently has anything to do with using a TPM, the original point was to use built-in solutions instead of third party ones since those more likely take advantage of hardware security features. Also I don't like how this implies that encrypting the operating system necessarily makes it more secure, this isn't always the case especially when the operating system is protected by a good secure boot or verified boot.

> Full disk encryption (FDE) is a comprehensive data encryption solution, encompassing the operating system and system files. FDE principally leverages hardware security features, such as a [secure cryptoprocessor](https://en.wikipedia.org/wiki/Secure_cryptoprocessor) (e.g., a [TPM](https://en.wikipedia.org/wiki/Trusted_Platform_Module)). Therefore, we recommend using the built-in FDE solutions for your operating system. For external drives, however, we still recommend [cross-platform tools](#multi-platform) for additional flexibility and to avoid vendor lock-in.

Reviewer 7: ```suggestion
Built-in OS encryption solutions generally leverage hardware security features, such as a [secure cryptoprocessor](https://en.wikipedia.org/wiki/Secure_cryptoprocessor). Therefore, we recommend using the built-in encryption solutions for your operating system. For cross-platform encryption, we still recommend [cross-platform tools](#multi-platform) for additional flexibility and to avoid vendor lock-in.
```

> Note the terms full *disk* encryption and full *volume* encryption are often used interchangeably.

Reviewer 8: These aren't the same though. Full disk implies the entire disk is encrypted and Full Volume encryption just means a volume on the disk is encrypted. I think we should probably avoid using either of these terms and just call it encryption.

### BitLocker

<div class="admonition recommendation" markdown>

![BitLocker logo](assets/img/encryption-software/bitlocker.png){ align=right }

**BitLocker**, a proprietary solution built into Windows, provides full volume encryption to protect data at rest. BitLocker [leverages](https://learn.microsoft.com/en-us/windows/security/hardware-security/tpm/how-windows-uses-the-tpm#tpm-overview) the Trusted Platform Module (TPM) for hardware-based security, provided one is present. BitLocker is only officially supported on Pro, Enterprise and Education editions of Windows.

BitLocker can be managed through the Control Panel: in the search box on the taskbar, type "Manage BitLocker" and then select it from the list of results.

[:octicons-info-16:](https://learn.microsoft.com/windows/security/information-protection/BitLocker/BitLocker-overview){ .card-link title=Documentation}

</details>

</div>

> To learn more about BitLocker and TPM, see "[Understanding BitLocker TPM Protection](https://blog.elcomsoft.com/2021/01/understanding-BitLocker-tpm-protection)" by ElcomSoft, a forensics company.

Reviewer 7: ```suggestion
```

> In addition to the TPM, BitLocker can lock the normal startup process until the user supplies a pin or inserts a USB flash drive containing a startup key. Preboot authentication is designed to prevent the encryption keys from being loaded to system memory without the trusted user supplying another authentication factor. This feature helps mitigate direct memory access and memory remanence attacks.

Reviewer 2: ```suggestion
In addition to the TPM, BitLocker can lock the normal startup process until the user supplies a pin or inserts a USB flash drive containing a startup key. Pre-boot authentication is designed to prevent the encryption keys from being loaded to system memory without the trusted user supplying another authentication factor. This feature helps mitigate direct memory access and memory remanence attacks.
```
https://en.m.wikipedia.org/wiki/Pre-boot_authentication

> In addition to the TPM, BitLocker can lock the normal startup process until the user supplies a pin or inserts a USB flash drive containing a startup key. Pre-boot authentication is designed to prevent the encryption keys from being loaded to system memory without the trusted user supplying another authentication factor. This feature helps mitigate direct memory access and memory remanence attacks.

Reviewer 2: ```suggestion
In addition to the TPM, BitLocker can lock the normal startup process until the user supplies a PIN or inserts a USB flash drive containing a startup key. Pre-boot authentication is designed to prevent the encryption keys from being loaded to system memory without the trusted user supplying another authentication factor. This feature helps mitigate direct memory access and memory remanence attacks.
```

<details class="tip" markdown>

> <summary>Enabling preboot authentication</summary>

Reviewer 2: ```suggestion
<summary>Enabling pre-boot authentication</summary>
```

1. Enable BitLocker on your operating system drive.

> 2. Allow for preboot authentication:

Reviewer 2: ```suggestion
2. Allow for pre-boot authentication:
```
https://en.m.wikipedia.org/wiki/Pre-boot_authentication

    1. Open the Group Policy Editor and navigate to:<br>
    **Computer Configuration** → **Administrative Templates** → **Windows Components** → **BitLocker Drive Encryption** → **Operating System Drives**
    2. Open and enable the setting "**Require Additional Authentication at Startup**."
    3. Under "**Configure TPM Startup PIN**," select: "**Require Startup PIN With TPM**."
3. Add a pin:
    1. As an administrator, open a command prompt window.
    2. Type and run:<br>
        `manage-bde -protectors -add c: -TPMAndPIN`<br>
        If necessary, replace `c:` with the drive letter of your operating system drive.
    3. Type and confirm a pin of 6 to 20 characters (you won't see the characters as you type them).
4. Check the preboot authentication status:
    1. Type and run: `manage-bde -status`.

>     2. Check whether "**TPM and Pin**" is listed under "**Key Protectors**."

Reviewer 2: ```suggestion
    2. Check whether "**TPM and PIN**" is listed under "**Key Protectors**."
```
PIN: https://en.m.wikipedia.org/wiki/Personal_identification_number

- You can change your preboot pin at any time by running: `manage-bde -changepin c:`, again, replacing `c:` with the drive letter of your operating system drive.

> - To remove the preboot pin:

Reviewer 2: ```suggestion
- To remove the pre-boot pin:
```

    1. Configure the Group Policy "**Configure TPM Startup PIN**" back to "**Allow Startup PIN With TPM**".
    2. Run: `manage-bde -protectors -add c: -TPM`.

</details>

> While BitLocker is not officially supported on Windows Home, it can be enabled on Home editions with a few extra steps.

Reviewer 2: ```suggestion
While BitLocker is not officially supported on Windows Home, it can be enabled with a few extra steps.
```
I removed that part since it's just a repetition on the same sentence.

<details class="example" markdown>
<summary>Enabling BitLocker on Windows Home</summary>

#### Preliminary checks

- You must have partitions formatted with a [GUID Partition Table](https://en.wikipedia.org/wiki/GUID_Partition_Table).
- You must have a TPM (v1.2, 2.0+) and it must be enabled.

> - You may need to [disable the non-Bitlocker "Device encryption" functionality](https://discuss.privacyguides.net/t/enabling-bitlocker-on-the-windows-11-home-edition/13303/5) if it is enabled.

Reviewer 2: ```suggestion
- You may need to [disable the non-BitLocker "Device encryption" functionality](https://discuss.privacyguides.net/t/enabling-bitlocker-on-the-windows-11-home-edition/13303/5) if it is enabled.
```
Just added a capital L to "BitLocker" to stay consistant on this page. Also, Microsoft writes their product name with that capital L.

![FileVault logo](assets/img/encryption-software/filevault.png){ align=right }

> **FileVault**, a proprietary solution integrated into macOS, offers on-the-fly volume encryption to secure all data at rest. On Macs with Apple silicon or a T2 Security Chip, FileVault [leverages](https://support.apple.com/guide/security/volume-encryption-with-filevault-sec4c6dc1b6e/web) the Secure Enclave and its hardware security capabilities.

Reviewer 2: ```suggestion
**FileVault**, a proprietary solution integrated into macOS, offers on-the-fly volume encryption to secure all data at rest. On Macs with Apple Silicon or a T2 Security Chip, FileVault [leverages](https://support.apple.com/guide/security/volume-encryption-with-filevault-sec4c6dc1b6e/web) the [Secure Enclave](https://support.apple.com/en-us/guide/security/sec59b0b31ff/web) and its hardware security capabilities.
```

Reviewer: Apple doesn't capitalize "silicon" in their own documentation

Reviewer 2: ```suggestion
**FileVault**, a proprietary solution integrated into macOS, offers on-the-fly volume encryption to secure all data at rest. On Macs with Apple silicon or a T2 Security Chip, FileVault [leverages](https://support.apple.com/guide/security/volume-encryption-with-filevault-sec4c6dc1b6e/web) the [Secure Enclave](https://support.apple.com/en-us/guide/security/sec59b0b31ff/web) and its hardware security capabilities.
```
